### PR TITLE
[QAA-312][DETOX] Speculos add accounts tests

### PR DESCRIPTION
**Pull request by abdurrahman-ledger**
### ✅ Checklist

- [ ] `npx changeset` was attached. (no need)
- [X] **Covered by automatic tests.** 
- [X] **Impact of the changes:** 
  - LLM E2E Tests

### 📝 Description

Improving LLM Detox x Speculos :

- Disable speculos iOS Speculos tests as not supported on CI yet
- Making enums LLD common to use in LLM
- Creating a DETOX env to decouple flakiness related changes from MOCKed tests
- Fixing speculos destroy to wait for it before continuing
- Improving Speculos add account test
- Separate each coin by file to enable parallelisation
- Fixing iOS caching to take into account .env.mock file

### ❓ Context

- [PR checks](https://github.com/LedgerHQ/ledger-live/actions/runs/11616670591?pr=8186) with correct workflow
- [Speculos tests](https://github.com/LedgerHQ/ledger-live/actions/runs/11605794644/job/32323224549)
- [LLD E2E Speculos tests](https://github.com/LedgerHQ/ledger-live/actions/runs/11609750577)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: .github/workflows/build-and-test-pr.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
     name: "Test Mobile E2E"
     needs: determine-affected
     if: ${{contains(needs.determine-affected.outputs.paths, 'ledger-live-mobile') && !github.event.pull_request.head.repo.fork}}
-    uses: LedgerHQ/ledger-live/.github/workflows/test-mobile-e2e-reusable.yml@develop
+    uses: LedgerHQ/ledger-live/.github/workflows/test-mobile-e2e-reusable.yml@support/qaa_312_detox_speculos_add_accounts
```
Comment (bharamboure-ledger) [inline .github/workflows/build-and-test-pr.yml:57]:
```suggestion
    uses: LedgerHQ/ledger-live/.github/workflows/test-mobile-e2e-reusable.yml@develop
```

File: .github/workflows/test-mobile-e2e.yml [hunk 2]
```diff
@@ -55,5 +55,5 @@ permissions:
 jobs:
   test-mobile-e2e:
     name: "Test Mobile E2E"
-    uses: LedgerHQ/ledger-live/.github/workflows/test-mobile-e2e-reusable.yml@develop
+    uses: LedgerHQ/ledger-live/.github/workflows/test-mobile-e2e-reusable.yml@support/qaa_312_detox_speculos_add_accounts
```
Comment (bharamboure-ledger) [inline .github/workflows/test-mobile-e2e.yml:58]:
```suggestion
    uses: LedgerHQ/ledger-live/.github/workflows/test-mobile-e2e-reusable.yml@develop
```

Comment (abdurrahman-ledger) [reply .github/workflows/test-mobile-e2e.yml:58]: I am just waiting for the checks to pass before reverting last commit

File: .github/workflows/build-and-test-pr.yml (hunk 1 above)
Comment (abdurrahman-ledger) [reply .github/workflows/build-and-test-pr.yml:57]: I am just waiting for the checks to pass before reverting last commit

**Comment by abdurrahman-ledger** [comment]
[PR checks](https://github.com/LedgerHQ/ledger-live/actions/runs/11609904689?pr=8186) are all green with correct workflow from this branch

**Review by bharamboure-ledger — DISMISSED** [review]

File: apps/ledger-live-mobile/e2e/page/accounts/addAccount.drawer.ts [hunk 3]
```diff
@@ -47,4 +60,26 @@ export default class AddAccountDrawer {
...
+    const accountId = (await getIdOfElement(this.accountCardRegExp(), 0)).replace(
+      /^account-card-/,
+      "",
+    );
+    await this.finishAccountsDiscovery();
+    await this.tapSuccessCta();
+    return accountId;
+  }
+
+  async addAccount(currency: Currency) {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/page/accounts/addAccount.drawer.ts:78]: this method is not in use anywhere, is it expected?

File: apps/ledger-live-mobile/e2e/page/accounts/account.page.ts [hunk 4]
```diff
@@ -1,43 +1,40 @@
...
+  accountBalance = (accountId: string) => getElementById(`account-balance-${accountId}`);
+  accountSettingsButton = () => getElementById("account-settings-button");
+  accountAdvancedLogRow = () => getElementById("account-advanced-log-row");
+  operationHistorySectionId = (accountId: string) => `operations-history-${accountId}`;
+  accountScreenScrollView = "account-screen-scrollView";
+  accountAdvancedLogsId = "account-advanced-logs";
 
-  async expectAccountBalanceVisible() {
-    await expect(this.assetBalance()).toBeVisible();
+  async openAccountSettings() {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/page/accounts/account.page.ts:14]: do you think we need to add @step above each method to have the same level of detail in Allure report?

File: apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccount.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      await app.portfolio.waitForPortfolioPageToLoad();
+    });
+
+    $TmsLink(tmsLink);
+    it(`Perform an add account`, async () => {
+      await app.addAccount.openViaDeeplink();
+      await app.common.performSearch(currency.name);
+      await app.addAccount.selectCurrency(currency.currencyId);
+
+      deviceNumber = await app.common.addSpeculos(currency.speculosApp.name);
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccount.ts:20]: can we start speculos in before all hook?

File: apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountADA.spec.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+import { runAddAccountTest } from "./AddAccount";
+import { Currency } from "@ledgerhq/live-common/e2e/enum/Currency";
+
+runAddAccountTest(Currency.ADA, "B2CQA-2500, B2CQA-2650, B2CQA-2678");
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountADA.spec.ts:4]: do we need separate spec file for each currency? wouldn't it makes sense to do it in one file?

File: apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountALGO.spec.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+import { runAddAccountTest } from "./AddAccount";
+import { Currency } from "@ledgerhq/live-common/e2e/enum/Currency";
+
+runAddAccountTest(Currency.ALGO, "B2CQA-2497, B2CQA-2653, B2CQA-2681");
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountALGO.spec.ts:4]: when your PR is merged can you please update all affected tickets from regression scope to have "Automated on:" "LLM (Detox)"? or you maybe can do it even now

File: apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccount.ts (hunk 5 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccount.ts:20]: I am investigating this in my following ticket with the use of CLI.

File: apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountADA.spec.ts (hunk 6 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountADA.spec.ts:4]: Unfortunately, it's mandatory as Jest and Detox don't [parallelise](https://stackoverflow.com/questions/39831961/are-tests-inside-one-file-run-in-parallel-in-jest) tests in same files

File: apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountALGO.spec.ts (hunk 7 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountALGO.spec.ts:4]: Sure

File: apps/ledger-live-mobile/e2e/page/accounts/addAccount.drawer.ts (hunk 3 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/accounts/addAccount.drawer.ts:78]: Good catch, it is a testing remainder.

File: apps/ledger-live-mobile/e2e/page/accounts/account.page.ts (hunk 4 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/accounts/account.page.ts:14]: Sure, I will add them

File: apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountADA.spec.ts (hunk 6 above)
Comment (ypolishchuk-ledger) [reply apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountADA.spec.ts:4]: is it a problem though? we have the same issue with api tests where we use Jest and we don't mind it, the impact is not critical

File: apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccount.ts (hunk 5 above)
Comment (ypolishchuk-ledger) [reply apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccount.ts:20]: ok, thanks!

File: apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountADA.spec.ts (hunk 6 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountADA.spec.ts:4]: Testing time for all currencies in parallel in currently around 15min

File: apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccount.ts (hunk 5 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccount.ts]: I did the changes, as adding CLI commands seems more complicated than anticipated

File: apps/ledger-live-mobile/e2e/page/accounts/account.page.ts (hunk 4 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/accounts/account.page.ts:14]: Done

File: apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountADA.spec.ts (hunk 6 above)
Comment (ypolishchuk-ledger) [reply apps/ledger-live-mobile/e2e/specs/speculos/addAccount/AddAccountADA.spec.ts:4]: hm, ok let's have a ticket to investigate how we can improve parallelisation

**Review by ypolishchuk-ledger — DISMISSED** [review]

**Comment by abdurrahman-ledger** [comment]
[PR Checks](https://github.com/LedgerHQ/ledger-live/actions/runs/11616670591?pr=8186) are all green with correct LLM iOS E2E Tests workflow from this branch

**Review by ypolishchuk-ledger — APPROVED** [review]

**Review by bharamboure-ledger — APPROVED** [review]

**Comment by abdurrahman-ledger** [comment]
As Add account flow will be changed on LLM, decoupling into two PR:
https://github.com/LedgerHQ/ledger-live/pull/8384
https://github.com/LedgerHQ/ledger-live/pull/8386

**Pull request closed without merging**